kiwi/system/kernel.py

```diff
@@ -127,3 +129,30 @@ def copy_xen_hypervisor(self, target_dir, file_name=None):
                 [target_dir, '/', file_name]
             )
             Command.run(['cp', xen.filename, target_file])
+
+    def _setup_kernel_names_for_lookup(self):
+        """
+        The kernel image name is different per arch and distribution
+        This method returns a list of possible kernel image names in
+        order to search and find one of them
+
+        :rtype: list
+        :return: list of kernel image names
+        """
+        for kernel_dir in os.listdir(''.join([self.root_dir, '/lib/modules'])):
+            return [
+                # lookup the link names first, or the result from
+                # functions.sh::suseStripKernel() if a kiwi initrd
+                # is used
+                'vmlinux', 'vmlinuz', 'zImage',
+
+                # not found, then lookup the real kernel image names
+                # depending on the arch and os they are different
+                ''.join(['uImage-', kernel_dir]),
+                ''.join(['Image-', kernel_dir]),
+                ''.join(['zImage-', kernel_dir]),
+                ''.join(['vmlinuz-', kernel_dir, '.gz']),
+                ''.join(['vmlinux-', kernel_dir]),
+                ''.join(['image-', kernel_dir]),
+                ''.join(['vmlinuz-', kernel_dir])
+            ]
```

Comment:
Why "for loop" if it will work only once? — the `return` will kick first in line and that's it. Additionally, we could make it not like Assembler programming :smile: :
```python
patt = '{prefix}-{kdir}'
kdir = os.listdir(''.join([self.root_dir, '/lib/modules']))
out = ['vmlinux', 'vmlinuz', 'zImage']
if kdir:
    for prefix in ['uImage', 'Image', 'zImage', 'vmlinuz', 'vmlinux', 'image']:
        out.append(patt.format(prefix=prefix, kdir=kdir[0]))  # Same functionality as above
        if prefix == 'vmlinuz':
            out.append(out[-1] + '.gz')  # Repeat it again with the ".gz" suffix

return out
```

Not tested, but something like this.